components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/ApiDAOImpl.java

```diff
@@ -763,10 +763,178 @@ public String getLastUpdatedTimeOfEndpoint(String endpointId) throws APIMgtDAOEx
      */
     @Override
     public Comment getCommentByUUID(String commentId, String apiId) throws APIMgtDAOException {
+        final String query = "SELECT COMMENT_ID, COMMENT_TEXT, USER_IDENTIFIER, API_ID, "
+                + "CREATED_BY, CREATED_TIME, UPDATED_BY, LAST_UPDATED_TIME "
+                + "FROM AM_API_COMMENTS WHERE COMMENT_ID = ? AND API_ID = ?";
+
+        try (Connection connection = DAOUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            try {
+                statement.setString(1, commentId);
+                statement.setString(2, apiId);
+                statement.execute();
+                try (ResultSet rs = statement.getResultSet()) {
+                    if (rs.next()) {
+                        return constructCommentFromResultSet(rs);
+                    }
+                }
+            } catch (SQLException e) {
+                connection.rollback();
+                String errorMessage =
+                        "Error while retrieving comment for comment id: " + commentId + " and api id: " + apiId;
+                log.error(errorMessage, e);
+                throw new APIMgtDAOException(e);
```

Comment:
Yes, you are correct. Sorry about the confusion. We will add custom exceptions in Impl.